src/unimport/main.py

```diff
@@ -1,90 +1,146 @@
+                + color.paint(
+                    path.as_posix(), color.GREEN, self.config.use_color
+                )
             )
-            is_syntax_error = True
-            continue
+            if self.is_syntax_error is False:
+                self.is_syntax_error = True
 
-        unused_imports = list(
-            Import.get_unused_imports(config.include_star_import)
-        )
-        unused_import_names.update({imp.name for imp in unused_imports})
-        analysis.clear()
-
-        if config.check:
-            commands.check(path, unused_imports, config.use_color)
-        if any((config.diff, config.remove)):
-            refactor_result = refactor_string(
-                source=source, unused_imports=unused_imports
-            )
-            if config.diff:
-                exists_diff = commands.diff(path, source, refactor_result)
-                if config.permission and exists_diff:
-                    commands.permission(
-                        path,
-                        encoding,
-                        newline,
-                        refactor_result,
-                        config.use_color,
-                    )
-            if config.remove and source != refactor_result:
-                commands.remove(
-                    path, encoding, newline, refactor_result, config.use_color
+        try:
+            yield
+        finally:
+            analysis.clear()
+
+    def get_results(self) -> Iterator[_Result]:
+        for path in self.config.get_paths():
+            source, encoding, newline = utils.read(path)
+
+            with self.analysis(source, path):
+                unused_imports = list(
+                    Import.get_unused_imports(self.config.include_star_import)
                 )
-                refactor_applied = True
-
-    if not unused_import_names and config.check:
-        print(
-            color.paint(
-                f"{emoji.STAR} Congratulations there is no unused import in your project. {emoji.STAR}",
-                color.GREEN,
-                config.use_color,
-            )
+                if self.is_unused_imports is False:
+                    self.is_unused_imports = unused_imports != []
+
+                yield _Result(unused_imports, path, source, encoding, newline)
+
+    def check(self, result: _Result) -> None:
+        commands.check(
+            result.path, result.unused_imports, self.config.use_color
         )
 
-    return return_exit_code(
-        is_unused_import_names=bool(unused_import_names),
-        is_syntax_error=is_syntax_error,
-        refactor_applied=refactor_applied,
-    )
+    def remove(self, result: _Result, refactor_result):
+        commands.remove(
+            result.path,
+            result.encoding,
+            result.newline,
+            refactor_result,
+            self.config.use_color,
+        )
+        if self.refactor_applied is False:
```

Comment:
There is no need to if statement. You can use ```self.refactor_applied = True```

Reply:
Fix it